content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -126,9 +126,70 @@ TODO.
 
 ## Type-safe labels for the `RenderGraph`
 
-<div class="release-feature-authors">authors: @TODO</div>
-
-TODO.
+<div class="release-feature-authors">authors: @DasLixou</div>
+
+Bevy uses Rust's type system at many points like for `SystemSets` which gives it a modern feel.
+But this didn't apply to Bevy's render graph. In the render graph, hard-coded—and potentially overlapping—strings were used to define nodes and sub-graphs.
+
+```rust
+// Before 0.13
+impl MyRenderNode {
+    pub const NAME: &'static str = "my_render_node"
+}
+```
+
+We are now introducing a more robust way to name rendernodes and -graphs with the help of Rust's type system and Bevy's label system.
+
+```rust
+// After 0.13
+#[derive(Debug, Hash, PartialEq, Eq, Clone, RenderLabel)]
+pub struct MyRenderLabel;
+```
+
+With those, the long paths for const-values also get shortened and cleaner which results into this
+
+```rust
+// Before 0.13
+render_app
+    .add_render_graph_node::<ViewNodeRunner<MyRenderNode>>(
+        core_3d::graph::NAME,
+        MyRenderNode::NAME,
+    )
+    .add_render_graph_edges(
+        core_3d::graph::NAME,
+        &[
+            core_3d::graph::node::TONEMAPPING,
+            MyRenderNode::NAME,
+            core_3d::graph::node::END_MAIN_PASS_POST_PROCESSING,
+        ],
+    );
+
+// After 0.13
+use bevy::core_pipeline::core_3d::graph::{Labels3d, SubGraph3d};
+
+render_app
+    .add_render_graph_node::<ViewNodeRunner<MyRenderNode>>(
+        SubGraph3d,
+        MyRenderLabel,
+    )
+    .add_render_graph_edges(
+        SubGraph3d,
+        (
+            Labels3d::Tonemapping,
+            MyRenderLabel,
+            Labels3d::EndMainPassPostProcessing,
+        ),
+    );
+```
+
+When you need dynamic labels for render nodes, those can still be achieved via e.g. tuple structs:
+
+```rust
+#[derive(Debug, Hash, PartialEq, Eq, Clone, RenderLabel)]
+pub struct MyDynamicLabel(&'static str);
+```
+
+The nice thing is, that this don't has to be a string, it can be any type, also numbers etc.
```

Comment:
```suggestion
This is particularly nice because we don't have to store strings in here: we can use integers, custom enums or any other hashable type.
```